src/main/kotlin/org/rust/lang/core/completion/RsPrimitiveTypeCompletionProvider.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.completion
+
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionProvider
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.patterns.ElementPattern
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.PsiElement
+import com.intellij.util.ProcessingContext
+import org.rust.lang.core.psi.RsTypeReference
+import org.rust.lang.core.types.ty.*
+import org.rust.lang.core.withSuperParent
+import org.rust.stdext.buildList
+
+object RsPrimitiveTypeCompletionProvider : CompletionProvider<CompletionParameters>() {
+
+    private val primitives: List<String> = buildList {
+        addAll(TyInteger.NAMES)
+        addAll(TyFloat.NAMES)
+        add(TyBool.name)
+        add(TyStr.name)
+        add(TyChar.name)
+    }
+
+    val elementPattern: ElementPattern<PsiElement> get() =
+        PlatformPatterns.psiElement().withSuperParent<RsTypeReference>(3)
+
+    override fun addCompletions(parameters: CompletionParameters, context: ProcessingContext, result: CompletionResultSet) {
+        primitives.forEach {
+            result.addElement(LookupElementBuilder.create(it).bold())
```

Comment:
Let's set a higher priority on lookup elements to group all primitive types and avoid this situation:
<img width="500" alt="example" src="https://user-images.githubusercontent.com/6079006/54307125-f1e54680-45db-11e9-9dd0-6d32169973f8.png">